src/components/Warnings/WarningItem.jsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-shadow */
+import { useState } from 'react';
+
+import './Warnings.css';
+import WarningIcons from './WarningIcons';
+import WarningsModal from './WarningsModal';
+
+function WarningItem({
+  warningText,
+  handlePostWarningDetails,
+  warnings,
+  username,
+  handleDeleteWarning,
+  submitWarning,
+}) {
+  const [toggleModal, setToggleModal] = useState(false);
+  const [warning, setWarning] = useState(null);
```

Comment:
Perhaps grouping these in a state object could be an improvement as they all deal with the warning state?
`const [warningState, setWarningState] = useState({
  warning: null,
  deleteWarning: false,
  warningId: null,
});`